### PR TITLE
07309 rate limited logger

### LINKED ISSUE #7309: Rate Limited Logging Utility
Create a utility that allows us to rate limit logs that have the potential of being spammy.

### PR DESCRIPTION
**Pull request by cody-littley**
Closes #7309

### CODE REVIEW

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/utility/throttle/RateLimitedLogger.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+     * @param message   the message to write
+     * @param varargs   optional arguments to pass to the log4j logger
+     */
+    public void debug(
+            @NonNull final Marker logMarker, @NonNull final String message, @Nullable final Object... varargs) {
+
+        try (final Locked l = lock.lock()) {
+            final long deniedRequests = rateLimiter.getDeniedRequests();
+            if (rateLimiter.requestAndTrigger()) {
+                logger.debug(logMarker, generateMessage(message, deniedRequests), varargs);
```
Comment (hendrikebbers) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/utility/throttle/RateLimitedLogger.java:106]: You can use `log(Level level, Marker marker, String message, Object... params);` By doing so you need to write the method implementation only once.

**Review by hendrikebbers — COMMENTED** [review]
As already said I see this as a temporary solution that we should get rid of in future.

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/utility/throttle/RateLimitedLogger.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+     *
+     * @param level     the log level
+     * @param logMarker the marker to use
+     * @param message   the message to write
+     * @param varargs   optional arguments to pass to the log4j logger
+     */
+    public void log(
+            @NonNull final Level level,
+            @NonNull final Marker logMarker,
+            @NonNull final String message,
```
Comment (edward-swirldslabs) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/utility/throttle/RateLimitedLogger.java:105]: Do these arguments marked as `NonNull` need `Objects.requireNonNull` checks?

**Review by edward-swirldslabs — DISMISSED** [review]

**Review by lpetrovic05 — APPROVED** [review]

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/utility/throttle/RateLimitedLogger.java (hunk 2 above)
Comment (cody-littley) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/utility/throttle/RateLimitedLogger.java:105]: added

**Review by timo0 — APPROVED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

**Pull request merged**